gpflowopt/bo.py

```diff
@@ -166,16 +167,20 @@ def _create_bo_result(self, success, message):
         # Filter on constraints
         valid = self.acquisition.feasible_data_index()
 
-        if not np.any(valid):
-            return OptimizeResult(success=False,
-                                  message="No evaluations satisfied the constraints")
+        # Extract the samples that satisfies all constraints
+        if np.any(valid):
+            valid_X = X[valid, :]
+            valid_Y = Y[valid, :]
+        else:
+            success = False
+            message = "No evaluations satisfied the constraints"
 
-        valid_X = X[valid, :]
-        valid_Y = Y[valid, :]
+        # Split between objectives and constraints
         valid_Yo = valid_Y[:, self.acquisition.objective_indices()]
```

Comment:
Won't work if no feasible samples are found. Will fix, but I also found that EI won't in any case if nothing is feasible (came across this before).

Also constraint information that is displayed in the print is the prediction of the model. Probably also important